src/linux/mgsocket.c

```diff
@@ -50,11 +55,18 @@ int mg_socket_create_handle_error(int sock, mg_session *session) {
 int mg_socket_connect(int sock, const struct sockaddr *addr,
                       socklen_t addrlen) {
   long status = MG_RETRY_ON_EINTR(connect(sock, addr, addrlen));
-  if (status == -1L) {
+  if (status == -1L && errno != EINPROGRESS) {
```

Comment:
Add this only for `__EMSCRIPTEN__`

Reply:
I would either introduce a function to check the status and it would be defined differently based on EMSCRIPTEN is used or not, or put a comment about this to the source.